odxtools/structures.py

```diff
@@ -301,16 +307,29 @@ def parameter_dict(self) -> ParameterDict:
         })
         return param_dict
 
-    def _resolve_references(self,
+    def _build_odxlinks(self):
+        result = super()._build_odxlinks()
+
+        for param in self.parameters:
+            result.update(param._build_odxlinks())
+
+        return result
+
+    def _resolve_references(self, # type: ignore[override]
                             parent_dl: "DiagLayer",
-                            odxlinks: OdxLinkDatabase):
+                            odxlinks: OdxLinkDatabase) \
+            -> None:
         """Recursively resolve any references (odxlinks or sn-refs)
         """
+        super()._resolve_references(odxlinks)
+
         for p in self.parameters:
             if isinstance(p, ParameterWithDOP):
                 p.resolve_references(parent_dl, odxlinks)
-            if isinstance(p, TableKeyParameter):
+            elif isinstance(p, TableKeyParameter):
                 p.resolve_references(parent_dl, odxlinks)
+            else:
+                p._resolve_references(odxlinks)
```

Comment:
it would be better to make this a common function in base class of parameters

Reply:
this is a hack to be able to resolve SN-REFs: because to be able to do so in the limited way that mattered for us, we needed to access the parent diaglayer for those types of parameters where they are commonly used. I intend to be resolve SN-REFs properly at some point (probably by extending the `OdxLinkDatabase` class), but that is another major refactoring and thus left out of this PR for now. (in the mean time, naming the `_resolve_references()` method differently for such classes would significantly reduce the confusion?)

Reply:
add resolve_references(parent_dl, odxlinks) to parameter class, default implementation does nothing, and override it in sub classes.